pkg/apis/sensor/v1alpha1/types.go

```diff
@@ -823,50 +817,50 @@ type GitArtifact struct {
 	// Remote to manage set of tracked repositories. Defaults to "origin".
 	// Refer https://git-scm.com/docs/git-remote
 	// +optional
-	Remote *GitRemoteConfig `json:"remote" protobuf:"bytes,10,opt,name=remote"`
+	Remote *GitRemoteConfig `json:"remote,omitempty" protobuf:"bytes,10,opt,name=remote"`
 }
 
 // GitRemoteConfig contains the configuration of a Git remote
 type GitRemoteConfig struct {
 	// Name of the remote to fetch from.
-	Name string `json:"name" protobuf:"bytes,1,name=name"`
-	// +listType=urls
+	Name string `json:"name" protobuf:"bytes,1,opt,name=name"`
+
 	// URLs the URLs of a remote repository. It must be non-empty. Fetch will
 	// always use the first URL, while push will use all of them.
 	URLS []string `json:"urls" protobuf:"bytes,2,rep,name=urls"`
 }
 
 // GitCreds contain reference to git username and password
 type GitCreds struct {
-	Username *corev1.SecretKeySelector `json:"username" protobuf:"bytes,1,opt,name=username"`
-	Password *corev1.SecretKeySelector `json:"password" protobuf:"bytes,2,opt,name=password"`
+	Username *corev1.SecretKeySelector `json:"username,omitempty" protobuf:"bytes,1,opt,name=username"`
+	Password *corev1.SecretKeySelector `json:"password,omitempty" protobuf:"bytes,2,opt,name=password"`
 }
 
 // Event represents the cloudevent received from a gateway.
 type Event struct {
-	Context *EventContext `json:"context" protobuf:"bytes,1,name=context"`
-	Data    []byte        `json:"data" protobuf:"bytes,2,name=data"`
+	Context *EventContext `json:"context,omitempty" protobuf:"bytes,1,opt,name=context"`
+	Data    []byte        `json:"data" protobuf:"bytes,2,opt,name=data"`
 }
 
 // EventContext holds the context of the cloudevent received from a gateway.
 type EventContext struct {
 	// ID of the event; must be non-empty and unique within the scope of the producer.
-	ID string `json:"id" protobuf:"bytes,1,name=id"`
+	ID string `json:"id" protobuf:"bytes,1,opt,name=id"`
 	// Source - A URI describing the event producer.
-	Source string `json:"source" protobuf:"bytes,2,name=source"`
+	Source string `json:"source" protobuf:"bytes,2,opt,name=source"`
 	// SpecVersion - The version of the CloudEvents specification used by the event.
-	SpecVersion string `json:"specversion" protobuf:"bytes,3,name=specVersion"`
+	SpecVersion string `json:"specversion" protobuf:"bytes,3,opt,name=specversion"`
 	// Type - The type of the occurrence which has happened.
-	Type string `json:"type" protobuf:"bytes,4,name=type"`
+	Type string `json:"type" protobuf:"bytes,4,opt,name=type"`
 	// DataContentType - A MIME (RFC2046) string describing the media type of `data`.
-	DataContentType string `json:"dataContentType" protobuf:"bytes,5,name=dataContentType"`
+	DataContentType string `json:"dataContentType" protobuf:"bytes,5,opt,name=dataContentType"`
```

Comment:
`datacontenttype `? @VaibhavPage - please confirm.

https://github.com/cloudevents/sdk-go/blob/master/v2/event/eventcontext_v03.go#L35

Reply:
this is unchanged in the PR - fix in another PR?

Reply:
Sounds good.